x-pack/legacy/plugins/maps/public/layers/styles/vector/components/legend/style_property_legend_row.js

```diff
@@ -134,17 +136,18 @@ export class StylePropertyLegendRow extends Component {
   }
 
   render() {
-    const { name, options, range } = this.props;
+
+    const { range, style } = this.props;
     if (this._isStatic()) {
       return null;
     }
 
     let header;
-    if (options.color) {
-      header = <ColorGradient colorRampName={options.color}/>;
-    } else if (name === 'lineWidth') {
+    if (style.getOptions().color) {
+      header = style.renderHeader();
+    } else if (style.getStyleName() === vectorStyles.LINE_WIDTH) {
       header = renderHeaderWithIcons(getLineWidthIcons());
```

Comment:
This is inconsisent in that you moved the render to style for `DynamicColorProperty` but not for `DynamicSizeProperty`. I think you should move `renderHeaderWithIcons` to the style instance since you moved it to the color style instance